src/Orchard.Web/Modules/Orchard.OpenId/Controllers/AccountController.cs

```diff
@@ -68,11 +69,12 @@ public class AccountController : Controller {
             }
 
             var membershipSettings = _membershipService.GetSettings();
+
             if (user != null &&
                 membershipSettings.EnableCustomPasswordPolicy &&
                 membershipSettings.EnablePasswordExpiration &&
                 _membershipService.PasswordIsExpired(user, membershipSettings.PasswordExpirationTimeInDays)) {
-                return RedirectToAction("ChangeExpiredPassword", new { username = user.UserName });
+                return RedirectToAction("ChangeExpiredPassword", "Account", new { Area = "Orchard.Users", username = user.UserName });
```

Comment:
But if the user is logged in with one of the OpenId providers, we can't change his/her password, because we don't own it!
So I think we need to implement custom logic to detect whether the user is logged in using a local account and then redirect to the change password action.